mlrun/api/initial_data.py

```diff
@@ -494,33 +497,75 @@ def _enrich_project_state(
             db.store_project(db_session, project.metadata.name, project)
 
 
+def _delete_default_marketplace_source(db_session: sqlalchemy.orm.Session):
+    """
+    Delete default marketplace source directly from db
+    """
+    # Not using db.delete_marketplace_source() since it doesn't allow deleting the default marketplace source.
+    default_record = (
+        db_session.query(MarketplaceSource)
+        .filter(MarketplaceSource.index == mlrun.api.schemas.last_source_index)
+        .one_or_none()
+    )
+    if default_record:
+        logger.info(f"Deleting default marketplace source {default_record.name}")
+        db_session.delete(default_record)
+        db_session.commit()
+    else:
+        logger.info("Default marketplace source not found")
+
+
+def _update_default_marketplace_source(
+    db: mlrun.api.db.sqldb.db.SQLDB,
+    db_session: sqlalchemy.orm.Session,
+    overwrite: bool = False,
+):
+    """
+    Updates default marketplace source in db.
+    overwrite=True will delete the current default marketplace source before creating a new one.
+    """
+    hub_source = mlrun.api.schemas.MarketplaceSource.generate_default_source()
+    # hub_source will be None if the configuration has marketplace.default_source.create=False
+    if hub_source:
+        if overwrite:
+            _delete_default_marketplace_source(db_session)
+        logger.info("Adding default marketplace source")
+        # Not using db.store_marketplace_source() since it doesn't allow changing the default marketplace source.
+        hub_record = db._transform_marketplace_source_schema_to_record(
+            mlrun.api.schemas.IndexedMarketplaceSource(
+                index=mlrun.api.schemas.marketplace.last_source_index,
+                source=hub_source,
+            )
+        )
+        db_session.add(hub_record)
+        db_session.commit()
+    else:
+        logger.info("Not adding default marketplace source, per configuration")
+
+
 def _add_default_marketplace_source_if_needed(
     db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
     try:
-        hub_marketplace_source = db.get_marketplace_source(
-            db_session, config.marketplace.default_source.name
+        hub_marketplace_source = (
+            db_session.query(MarketplaceSource)
+            .filter(
+                MarketplaceSource.index
+                == mlrun.api.schemas.marketplace.last_source_index
+            )
+            .one_or_none()
         )
     except mlrun.errors.MLRunNotFoundError:
         hub_marketplace_source = None
```

Comment:
This `try/execpt` is not needed anymore, since you changed the logic above to use a query with `one_or_none()` which will return `None` if the record doesn't exist.